Source/Core/Core/HW/VideoInterface.cpp

```diff
@@ -761,30 +774,30 @@ void Update(u64 ticks)
     EndField();
   }
 
-  for (UVIInterruptRegister& reg : m_InterruptRegister)
-  {
-    if (s_half_line_count + 1 == 2u * reg.VCT)
-    {
-      reg.IR_INT = 1;
-    }
-  }
+  // Move to the next half-line and potentially roll-over the count to zero. If we've reached
+  // the beginning of a new full-line, update the timer
 
   s_half_line_count++;
-
-  if (s_half_line_count > GetHalfLinesPerEvenField() + GetHalfLinesPerOddField())
+  if (s_half_line_count == GetHalfLinesPerEvenField() + GetHalfLinesPerOddField())
   {
-    s_half_line_count = 1;
-    s_half_line_of_next_si_poll = num_half_lines_for_si_poll;  // first results start at vsync
+    s_half_line_count = 0;
   }
 
-  if (s_half_line_count == GetHalfLinesPerEvenField())
+  if (!(s_half_line_count & 1))
   {
-    s_half_line_of_next_si_poll = GetHalfLinesPerEvenField() + num_half_lines_for_si_poll;
+    s_ticks_last_line_start = CoreTiming::GetTicks();
   }
 
-  if (s_half_line_count & 1)
+  // Check if we need to assert IR_INT. Note that the granularity of our current horizontal
+  // position is limited to half-lines.
+
+  for (UVIInterruptRegister& reg : m_InterruptRegister)
   {
-    s_ticks_last_line_start = CoreTiming::GetTicks();
+    u32 h_thresh = (reg.HCT > m_HTiming0.HLW) ? 1 : 0;
```

Comment:
No need to save on letters here; especially when your PR is meant to make it easier to reason about code ;)
```suggestion
    u32 horizontal_threshold = (reg.HCT > m_HTiming0.HLW) ? 1 : 0;
```
Not sure if thats the best name though (or whether my guesses of what `h` and `thresh` were meant to stand for were correct).